### PR TITLE
Create new whitelabel palette generation algorithm

### PR DESCRIPTION
**Pull request by BboyStatix**
## Ticket Link
https://toruslabs.atlassian.net/browse/PD-3317

## What does this PR do?
Creates a whitelabel color palette generation algorithm in openlogin-utils that can be imported.

## TODOs
- [ ] Analyze build size

### CODE REVIEW

File: packages/openlogin-utils/src/whitelabel.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        .saturate(((step + 1) / colorsAmount) * (saturation / 100))
+        .mix(Color(mixColor), ((colorsShiftAmount / 100) * (step + 1)) / colorsAmount)
+        .hex()
+    );
+  }
+
+  return colorsList;
+}
+
+function generateWhiteLabelTheme(primary: string) {
```
Comment (BboyStatix) [inline packages/openlogin-utils/src/whitelabel.ts:22]: 1. Does it make sense to put this function in openlogin-utils? Will it affect build size? If so can we code split it?
2. Should this function instead only return theme colors and we let the caller of the function apply the colors themselves?
Technically the only colors that would need to be returned would be the primary color palette as the other colors don't require other shades being generated by us.

**Review by arch1995 — APPROVED** [review]
lgtm

File: packages/openlogin-utils/src/whitelabel.ts (hunk 1 above)
Comment (chaitanyapotti) [reply packages/openlogin-utils/src/whitelabel.ts:22]: 1. Yes. it doesn't affect build size.
2. should be fine

**Review by chaitanyapotti — APPROVED** [review]
lgtm. will change in next branch

**Pull request merged**